jsonrpc/src/codec.rs

```diff
@@ -46,6 +66,11 @@ impl Encoder for JsonCodec {
             Err(err) => return Err(Error::Encode(err.to_string())),
         };
         let buf = msg.as_bytes();
+
+        if buf.len() >= self.max_size {
+            return Err(Error::InvalidMsg("Msg too large".to_string()));
```

Comment:
The message is misleading, the buffer is full but there might be no single message which is too large